src/neuroconv/datainterfaces/behavior/fictrac/fictracdatainterface.py

```diff
@@ -117,311 +120,126 @@ def add_to_nwbfile(
-            name="SpatialSeriesIntegratedMotion",
-            data=data,
-            unit="radians",
-            reference_frame=reference_frame,
-            description=description,
-        )
-
-        if write_timestamps:
-            spatial_seriess_kwargs["timestamps"] = timestamps
-        else:
-            spatial_seriess_kwargs["rate"] = rate
-            spatial_seriess_kwargs["starting_time"] = starting_time
-
-        spatial_series = SpatialSeries(**spatial_seriess_kwargs)
-        compass_direction_container.add_spatial_series(spatial_series)
-
-        # Add error in rotation delta
-        rotation_delta_error = fictrac_data_df["rotation_delta_error"].values
-        reference_frame = "lab"
-        description = "Error in rotation delta in radians from the lab frame"
-
-        spatial_seriess_kwargs = dict(
-            name="SpatialSeriesRotationDeltaError",
-            data=rotation_delta_error,
-            reference_frame=reference_frame,
-            unit="radians",
-            description=description,
-        )
-
-        if write_timestamps:
-            spatial_seriess_kwargs["timestamps"] = timestamps
-
-        else:
-            spatial_seriess_kwargs["rate"] = rate
-            spatial_seriess_kwargs["starting_time"] = starting_time
-
-        spatial_series = SpatialSeries(**spatial_seriess_kwargs)
-        compass_direction_container.add_spatial_series(spatial_series)
+        position_container = Position(name="FicTrac")
+
+        for data_dict in self.column_to_nwb_mapping.values():
+            spatial_seriess_kwargs = dict(
+                name=data_dict["spatial_series_name"],
+                description=data_dict["description"],
+                reference_frame=data_dict["reference_frame"],
+            )
+
+            columns_in_dat_file = data_dict["columns_in_dat_file"]
+            data = fictrac_data_df[columns_in_dat_file].to_numpy()
+            if self.radius is not None:
+                data = data * self.radius
+                units = "meters"
+            else:
+                units = "radians"
+
+            spatial_seriess_kwargs.update(data=data, unit=units)
+
+            if write_timestamps:
+                spatial_seriess_kwargs["timestamps"] = timestamps
+            else:
+                spatial_seriess_kwargs["rate"] = rate
+                spatial_seriess_kwargs["starting_time"] = starting_time
+
+            # Create the spatial series and add it to the container
+            spatial_series = SpatialSeries(**spatial_seriess_kwargs)
+            position_container.add_spatial_series(spatial_series)
 
         # Add the compass direction container to the processing module
         processing_module = get_module(nwbfile=nwbfile, name="Behavior")
-        processing_module.add_data_interface(compass_direction_container)
+        processing_module.add_data_interface(position_container)
+
+    column_to_nwb_mapping = spatial_series_descriptions = {
```

Comment:
It is a class attribute so the tab is playing trick with your eyes. It is not within the `add_to_nwb` method.

Reply:
I see - but shouldn't all the class attributes be defined in a single place up at the top?

Reply:
No such a rule as far as I am aware. Specially for long ones as this. I have a slight preference for having it after as I want the logic of the interface to be presented before the configuration of the mapping. That's a weak preference though, if you show me any sort of convention I am fine with sending it to the beginning. If your problem is aestethical, I think it looks better once the other methods are added in the https://github.com/catalystneuro/neuroconv/pull/607 PR.

Reply:
Ah, #607 that was just merged here. Great.

Reply:
I failed to notice the chain else I would have gotten this one in first 😆

Reply:
>  if you show me any sort of convention I am fine with sending it to the beginning.

First result found on google: https://www.digitalocean.com/community/tutorials/understanding-class-and-instance-variables-in-python-3#class-variables

> Defined outside of all the methods, class variables are, by convention, typically placed right below the class header and before the [constructor method](https://www.digitalocean.com/community/tutorials/how-to-construct-classes-and-define-objects-in-python-3#the-constructor-method) and other methods.

In this case I missed the fact that it was defined as such because it was interwoven between methods; makes more sense to me that all such variables should be defined in one nice visual cluster at the top as convention

Reply:
I mean something like a PEP or any big project style guide. I agree that people usually put attributes at the beginning of the class but usually those attributes not a big dictionary that covers ~100 lines. Practically beats purify and I weight readiability heavier than convention here.

But, the mistake of mine for not specifying the request well enough is on me. It is now on top.

Reply:
> Practically beats purify and I weight readiability heavier than convention here.

The convention is for readability - as the first person to review a PR in violation of the unstated convention, I failed to properly read it - hence evidence for the convention

Only a sample size of one, but you're welcome to open a broader discussion as we write up and centralize these practices into the organization-wide coding style guide